platform/linux-generic/odp_traffic_mngr.c

```diff
@@ -2918,9 +2919,11 @@ odp_tm_t odp_tm_create(const char            *name,
 	/* If we are using pktio output (usual case) get the first associated
 	 * pktout_queue for this pktio and fail if there isn't one.
 	 */
-	if (egress->egress_kind == ODP_TM_EGRESS_PKT_IO &&
-	    odp_pktout_queue(egress->pktio, &pktout, 1) != 1)
-		return ODP_TM_INVALID;
+	if (egress->egress_kind == ODP_TM_EGRESS_PKT_IO) {
+		rc = _odp_pktio_pktout_tm_config(egress->pktio, &pktout);
```

Comment:
I wonder if the TM API allows having more than one TM system attached to the same pktio egress. I could not immediately see anything in the current linux-gen implementation that would not work in that case (but I may have missed it). After this change there can be only one TM system for one pktio. I wonder if there should be an API change or API clarification about it.

The commit message of this patch needs improvement. It should describe better what is done and why (pktio output done by TM implementation is done in a way that requires an output queue to exists, but the ODP application cannot configure it when output mode is TM). The current message also says that this adds an internal API, but this patch also adds usage for that API.

Reply:
Ack, I broke that functionality. I believe API does allow multiple TM systems with same egress. Our platform however doesn't support it. Should I change the logic to do pktout queue setup in _odp_pktio_pktout_tm_config() only once ?